rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala

```diff
@@ -0,0 +1,225 @@
+import net.liftweb.http.Req
+import com.normation.rudder.web.rest.RestAPI
+import net.liftweb.json.JsonAST.JValue
+import com.normation.rudder.web.rest.RestExtractorService
+import com.normation.rudder.web.rest.RestUtils
+import com.normation.rudder.web.rest.ApiVersion
+import com.normation.rudder.appconfig._
+import net.liftweb.common.Full
+import net.liftweb.common.Failure
+import java.io.File
+import net.liftweb.json.JsonAST.JString
+import net.liftweb.json.JsonAST.JArray
+import java.nio.file.attribute.PosixFilePermissions
+import java.nio.file.Files
+import java.nio.file.Paths
+import org.joda.time.DateTime
+import com.normation.rudder.web.components.DateFormaterService
+import net.liftweb.json.JsonDSL
+import net.liftweb.json.JsonAST.JField
+import net.liftweb.json.JsonAST.JObject
+import net.liftweb.http.JsonResponse
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Loggable
+import java.nio.charset.StandardCharsets
+import net.liftweb.http.StreamingResponse
+import java.io.ByteArrayInputStream
+import java.nio.file.Path
+import net.liftweb.http.rest.RestHelper
+
+class SharedFilesAPI(
+    restExtractor    : RestExtractorService
+  , sharedFolderPath : String
+) extends RestHelper with Loggable {
+
+  def checkPathAndContinue(path : String)(fun : Path => Box[LiftResponse]) : Box[LiftResponse] = {
+    import net.liftweb.util.Helpers._
+    (tryo {
+      val sharedFile = Paths.get(sharedFolderPath).normalize()
+      val filePath = Paths.get(sharedFolderPath,path).normalize()
+      if (filePath.startsWith(sharedFile)) {
+        fun(filePath)
+      } else {
+        Failure("Unauthorized access")
+      }
+    }).flatMap { identity }
+  }
+
+  def serialize(file:File) : Box[JValue] = {
+    import net.liftweb.util.Helpers._
+    import net.liftweb.json.JsonDSL._
+    tryo{
+      val date = new DateTime(file.lastModified())
+      ( ("name"  -> file.getName)
+      ~ ("size"  -> file.length())
+      ~ ("type"  -> (if (file.isFile) "file" else "dir"))
+      ~ ("date"  -> DateFormaterService.getFormatedDate(date))
+      ~ ("perms" ->  PosixFilePermissions.toString(Files.getPosixFilePermissions(Paths.get(file.getPath))))
+      )
+    }
+  }
+
+  def errorResponse(message : String): LiftResponse = {
+      import net.liftweb.json.JsonDSL._
+      val content =
+        ( ("success" -> false)
+        ~ ("error"   -> message)
+        )
+      JsonResponse(content,Nil,Nil, 500)
+  }
+
+  def downloadFile(path : Path) : Box[LiftResponse] = {
+    if (Files.exists(path)) {
+      if (Files.isRegularFile(path)) {
+        val data = Files.readAllBytes(path)
+        val headers =
+         ("Content-type" -> "application/octet-stream") ::
+         ("Content-length" -> data.length.toString) ::
+         ("Content-disposition" -> s"attachment; filename=${path.getFileName}") ::
+         Nil
+
+        Full(StreamingResponse(new ByteArrayInputStream(data),() => {}, data.length, headers, Nil, 200))
+      } else {
```

Comment:
Here, you are reading the file into memory to then stream it to the network. You should side-step the memory part to avoid to consume huge quantity of RAM on ISO.
It's easy to get the file size without reading it with (new File(path)).length (perhaps there is an quivalent method on nio). 
And for the streaming part, a simple FileIntputStream should be ok:
val file = new File(path)
Full(StreamingResponse(new FileInputStream(file),() => {}, file.length, headers, Nil, 200))